examples/directml/llm/user_script.py

```diff
@@ -0,0 +1,224 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import gc
+import random
+
+import config
+import numpy as np
+import torch
+from datasets import load_dataset
+from decoder_model import DecoderModel
+from falcon import convert_falcon_weights
+from llava_model import LlavaModel
+from phi import convert_phi_weights
+from transformers import AutoConfig, AutoTokenizer
+
+
+# Helper latency-only dataloader that creates random tensors with no label
+class RandomDataLoader:
```

Comment:
>RandomDataLoader [](http://example.com/codeflow?start=6&length=16)

do we need limit the length of the dataloader?

Reply:
The length is set via the `calibration_sampling_size` parameter in llm.py.

Reply:
Please see the comments of https://github.com/microsoft/Olive/blob/main/olive/data/component/dataset.py#L113, which I hit it before.

Reply:
without the raise of IndexError, the for loop for the RandomDataloader will run infinitely. Not sure it is the expected behavior.